gensim/similarities/docsim.py

```diff
@@ -569,6 +569,145 @@ def __str__(self):
         return "%s<%i docs, %i features>" % (self.__class__.__name__, len(self), self.index.shape[1])
 #endclass MatrixSimilarity
 
+class WmdSimilarity(interfaces.SimilarityABC):
+    """
+    Document similarity (like MatrixSimilarity) that uses the negative of WMD
+    as a similarity measure. See gensim.models.word2vec.wmdistance for more
+    information.
+
+    When a `num_best` value is provided, only the most similar documents are
+    retrieved. In this case, an efficient algorithm called "prefetch and prune"
+    is used.
+
+    When using this code, please consider citing the following papers:
+    * Ofir Pele and Michael Werman, "A linear time histogram metric for improved SIFT matching".
+    * Ofir Pele and Michael Werman, "Fast and robust earth mover's distances".
+    * Matt Kusner et al. "From Word Embeddings To Document Distances".
+
+    Example:
+        # Given a document collection "corpus", train word2vec model.
+        model = word2vec(corpus)
+        instance = WmdSimilarity(corpus, model, num_best=10)
+
+        # Make query.
+        sims = instance[query]
+    """
+    def __init__(self, corpus, w2v_model, num_best=None, init_distances=True, chunksize=256):
+        """
+        corpus:             List of lists of strings, as in gensim.models.word2vec.
+        w2v_model:          A trained word2vec model.
+        num_best:           Number of results to retrieve. If provided, a fast algorithm
+                            called "prefetch and prune" is used.
+        init_distances:     Whether or not to initialize a distance matrix
+                            (euclidean distance between words in vocab).
+        """
+        self.corpus = corpus
+        self.w2v_model = w2v_model
+        self.num_best = num_best
+        self.chunksize = chunksize
+
+        # Normalize embeddings to sum to 1.
+        self.w2v_model.init_sims(replace=True)
+
+        # Normalization is not possible, as corpus is a list (of lists) of strings.
+        self.normalize = False
+
+        # index is simply an array from 0 to size of corpus.
+        self.index = numpy.array(range(len(corpus)))
```

Comment:
In `gensim.interfaces.SimilarityABC`, which `WmdSimilarity` extends. The Similarity base class uses `index` to sort the documents.